### ISSUE TITLE
Provide an installer download

### ISSUE THREAD

**Issue by thany** [File-New-Project/EarTrumpet#1004]
## Description
There is no installer to download. This is a perfectly fine, well-accepted, and vastly common way to install anything on Windows.

You might not like it, but users do. Please make a normal installer. Make it with Microsoft technology (MSI) if you must, but please make a regular ol' downloadable installer.

Winget and chocolatey are not good (to put it lightly). Not even for me as a developer, and certainly not for novice users. The Store doesn't even work everywhere (geographically speaking that is), isn't allowed by company policy, is blocked by a firewall, or is just plain broken. And no, any form of manual override to this is just as horrible as a CLI variety.

## Steps to reproduce
1. Open the homepage
2. Find the installer
3. Download it
4. Execute it
5. Profit

## Meta
* EarTrumpet version: none
* Windows 10 version: 21H2

## Screenshots
I'm not going to mock up an installer. Literally anything is better than what you have now.

**Comment by rootexpression** [comment]
Seconded.

**Comment by riverar** [comment]
Hey @thany @rootexpression, as mentioned in numerous other threads, we are not providing a standalone installer. To do so would require us to take on the burden of hosting the files, maintaining the installer, doing all the work of actually updating the files on disk, etc. in perpetuity. In other words, working on all that takes us away from working on the core app. Thankfully, these are all problems that the Microsoft Store, Windows Package Manager, and Chocolately have already solved and I can say over the course of 6 years now we've had minimal issues.

> The Store doesn't even work everywhere (geographically speaking that is)

The Microsoft Store is available in [over 200 countries](https://docs.microsoft.com/en-us/windows/uwp/publish/define-market-selection?WT.mc_id=WD-MVP-5002756). For users in countries that don't have access to the Microsoft Store, EarTrumpet is also available via Chocolatey and Winget, both of which can be downloaded and installed automatically with or without the choco/winget tooling.

Here's a recap of all the supported deployment options and can easily be automated:

* Microsoft Store (https://www.microsoft.com/p/eartrumpet/9nblggh516xp)
* Chocolatey (https://community.chocolatey.org/packages/eartrumpet)
* Windows Package Manager Client (https://github.com/microsoft/winget-pkgs/blob/master/manifests/f/File-New-Project/EarTrumpet/2.1.9.0/File-New-Project.EarTrumpet.yaml)
* AppInstaller (https://install.eartrumpet.app/master/EarTrumpet.Package.appinstaller)

For those that don't want updates or support, you can also grab EarTrumpet via one of these methods:

* Raw appxbundle (e.g. https://install.eartrumpet.app/master/EarTrumpet.Package_2.1.9.1_x86.appxbundle)
* Raw zip (e.g. https://community.chocolatey.org/api/v2/package/eartrumpet/2.1.9.0)

Hope that helps.

Closing as duplicate of #980 #840 etc.

**Comment by thany** [comment]
> take on the burden of hosting the files

Hosting the installer is a problem even though your are perfectly able to host the rest of the website? Seems like a non-issue. It would (or should) end up in Github's releases, and a link on the homepage. I don't see why that's so difficult.

> maintaining the installer

Like the currently available packages. So that's also a non-issue.

> doing all the work of actually updating the files on disk

The installer does that. Now you're just making excuses. You don't have to make an installer from scratch, any existing installer will do, including MSI.

> working on all that requires resources and takes us away from working on the core app

Only the one time. Once set up correctly, building an installer is part of the build process. Can be done fully automatically.

> Thankfully, these are all problems that...

They aren't problems unless you imagine them.

------
Please reopen this issue, make an installer and stop making excuses. It's embarassing really, iyam. Thousands of Windows programs are installed using an installer, and installers are NOT hard to make, maintain, and host. It is TOTALLY common to do it that way.

On top of that, MSI installers can be automated, many MANY users prefer an installer, and it's just simply the way Windows software is commonly installed.

**Comment by riverar** [comment]
Is there something preventing you from using the appxbundle above? It's the same thing.

**Comment by EmberHeartshine** [comment]
> Is there something preventing you from using the appinstaller/appxbundle above? It's the same thing.

As I mentioned in #840, expecting someone to install an entire framework to keep *one* program up to date is absurd. That's like saying, "just set up rpm on your Debian system, it's really easy!" While technically true, it's entirely counterproductive.

**Comment by GoldenTao** [comment]
> AppInstaller (https://install.eartrumpet.app/master/EarTrumpet.Package.appinstaller)

@EmberHeartshine Have you tried the appinstaller listed above in Raf's comment?  No framework required as per your requirements.

**Comment by EmberHeartshine** [comment]
> @EmberHeartshine Have you tried the appinstaller listed above in Raf's comment? No framework required as per your requirements.

The MSIX system that requires the WUP framework?

**Comment by thany** [comment]
> Is there something preventing you from using the appinstaller/appxbundle above? It's the same thing.

It's not. It requires a working Store and backing services. Not available everywhere (not only geographically, think enterprises/corporate scenarios, or just Microsoft bollocksing up again). And it only works in Windows 10. And it doesn't install where applications are expected to be installed (C:\Program Files) not does it allow a choice for that. And it can't install "for all users". And it's nowhere to be found, except when making a big deal out of the total absence of a sensible installer.

Look mate, just produce an installer and stop ambarassing yourselves. Everyone produces an installer, esp for GUI apps that aren't built on UWP.

**Comment by riverar** [comment]
It doesn't "require UWP".

**Comment by thany** [comment]
> It doesn't "require the UWP framework".

It does if you take this sorry excuse for an """"installer"""":

> AppInstaller (https://install.eartrumpet.app/master/EarTrumpet.Package.appinstaller)

**Comment by EmberHeartshine** [comment]
> It doesn't "require the UWP framework".

Uh, okay, [tell that to Microsoft](https://docs.microsoft.com/en-us/windows/msix/package/packaging-uwp-apps) then.

**Comment by thany** [comment]
Anyway, a normal installer is *just how it's done in the Windows world*. Period.

If you don't like it, stop making Windows software and move over to Linux. Package managers are the norm there. You'll _love_ it.

**Comment by riverar** [comment]
Our AppInstaller / MSIX bundles don't need the Microsoft Store or "the UWP framework" (not a real thing). But because there appears to be some ongoing confusion about each of these options, I'll work on more detailed installation documentation that covers installation of EarTrumpet on all consumer and enterprise versions of Windows.

I'd like to understand, however, how you are specifically blocked and how a EXE/MSI-based installer unblocks you. I must be missing something because we have 4 million users that have successfully installed EarTrumpet across both consumer and enterprise (incl. LTSB) versions of Windows 10 and Windows 11.

Thanks for your patience!

**Comment by EmberHeartshine** [comment]
If you're not going to read the documentation provided to you, I'm checking out of this conversation.

**Comment by riverar** [comment]
@EmberHeartshine I don't understand the point you're trying to make. You linked to an article on how to package an app with MSIX. Can you be more specific?

To give you a peek behind the scenes: EarTrumpet is Desktop Bridge (previously called Project Centennial) app. Whether or not these are considered "UWP apps" is a gray area so I won't call it that. But I was just pointing out that there's no such thing as a Universal Windows Platform Framework and there's no additional _UWP thing_ you need to install to support our app. For example, you can successfully install EarTrumpet on Long Term Servicing Branch editions of Windows 10 Enterprise with no problem.

**Comment by thany** [comment]
The fact that 4 million users (I don't believe that, but for the sake of argument, let's say it's correct) are putting up with the absence of an installer, doesn't mean that it isn't a problem. I just don't understand why you keep defending the poor choice of not having a proper installer. You really are embarassing yourself.

You'd do good to either say "sorry, you're right, every program has its installer, and so should ours" or silently do so anyway and make me look like a fool. Either option works for me.

But without an installer, you're putting no one but yourself on the fool's pedestal. Trust me, if you make an installer, your (and its) userbase will grow quite naturally.

**Comment by riverar** [comment]
Let's say we packaged the app with a magical technology that you approve of. How do you propose we keep these EarTrumpet users up to date?

**Comment by riverar** [comment]
Locking this thread as it's no longer productive.
